virtwho/register.py

```diff
@@ -1,5 +1,487 @@
-"""
-class SubscriptionManager()
-class RHSMAPI()
-class SatelliteCLI()
-"""
+import re
+import time
+
+from virtwho import logger, FailException
+from virtwho.configure import virtwho_ssh_connect
+from virtwho.configure import guest_ssh_connect
+from virtwho.configure import get_register_handler
+from virtwho.settings import config
+
+
+class SubscriptionManager:
+    """"""
+
+    def __init__(self, mode, register_type):
+        """
+        Define the global variables.
+        :param mode: the hypervisor mode.
+            (esx, xen, hyperv, rhevm, libvirt, kubevirt, local)
+        :param register_type: the subscription server. (rhsm, satellite)
+        """
+        self.mode = mode
+        self.register_type = register_type
+        self.ssh_virtwho = virtwho_ssh_connect(self.mode)
+        self.ssh_guest = guest_ssh_connect(self.mode)
+        self.rhsm_conf = '/etc/rhsm/rhsm.conf'
+        self.rhsm_conf_backup = '/backup/rhsm.conf'
+
+        self.register_section = get_register_handler(register_type)
+        self.server = self.register_section.server
+        self.username = self.register_section.username
+        self.password = self.register_section.password
+        self.port = self.register_section.port
+        self.prefix = self.register_section.prefix
+        self.default_org = self.register_section.default_org
+        if self.register_type == 'satellite':
+            self.secondary_org = self.register_section.secondary_org
+            self.activation_key = self.register_section.activation_key
+
+    def register(self, second_org=False, activ_key=False, guest=False):
```

Comment:
The function `register` should be supported to register system to both rhsm and satellite.
If we don't backup the rhsm.conf, is it possible to use subscription-manager command to register both rhsm/satellite by specific options, such as:

subscription-manager --serverurl  --username --password --org --env

Reply:
Updated the register() function to use command ```subscription-manager register --serverurl  --org  --username --password ``` to regiter host.
Didn't define the ```--baseurl``` by command because no need to set it for rhsm and satellite cert can configure it.